src/components/DisplayWeb3.tsx

```diff
@@ -13,51 +18,110 @@ interface ClientVersion {
+    // User State
+    const [account, setAccount] = useState<string>()
     const [balance, setBalance] = useState<string>()
-
     const [networkName, setNetworkName] = useState<string>()
-
     const [clientVersion, setClientVersion] = useState<ClientVersion>()
+    // Send
+    const [amountToSend, setAmountToSend] = useState<string>('0')
+    const [recipientAddress, setRecipientAddress] = useState<string>(null)
+    // Async
+    const [error, setError] = useState<Error>(null)
+    const [loading, setLoading] = useState<boolean>(false)
 
     useEffect(() => {
         const setter = async (): Promise<void> => {
-            const accountsProm = web3Api.getAccounts()
-            const networkNameProm = web3Api.getNetworkName()
-            const clientVersionProm = web3Api.web3.currentProvider.send('web3_clientVersion', [])
-
-            const [account] = await accountsProm
-
-            const [balance, networkName, clientVersion] = await Promise.all([
-                web3Api.getBalance(account),
-                networkNameProm,
-                clientVersionProm,
-            ])
-
-            ReactDOM.unstable_batchedUpdates(() => {
-                setAccount(account)
-                setBalance(balance)
-                setNetworkName(networkName)
-                setClientVersion(clientVersion)
-            })
+            try {
+                setError(null)
+                setLoading(true)
+
+                const accountsProm = web3Api.getAccounts()
+                const networkNameProm = web3Api.getNetworkName()
+                const clientVersionProm = web3Api.web3.currentProvider.send('web3_clientVersion', [])
+
+                const [account] = await accountsProm
+
+                const [balance, networkName, clientVersion] = await Promise.all([
+                    web3Api.getBalance(account),
+                    networkNameProm,
+                    clientVersionProm,
+                ])
+
+                ReactDOM.unstable_batchedUpdates(() => {
+                    setAccount(account)
+                    setBalance(balance)
+                    setNetworkName(networkName)
+                    setClientVersion(clientVersion)
+                })
+            } catch (error) {
+                console.error(error)
+                setError(error)
+            } finally {
+                setLoading(false)
+            }
         }
 
         setter()
     }, [web3Api])
 
+    const handleSendTransaction = async (): Promise<TransactionReceipt | void> => {
+        try {
+            // setError(null)
+            setLoading(true)
+            const txReceipt = await web3Api.web3.eth.sendTransaction({
+                from: account,
+                to: recipientAddress || DEFAULT_RECIPIENT,
+                value: toWei(amountToSend),
+            })
+            console.log('TX Receipt: ', txReceipt)
+
+            return txReceipt
+        } catch (error) {
```

Comment:
what's the point on rethrowing? You can do the finally without catch

Reply:
Unnecessary here, agree